// @aws-sdk/client-rds/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { RDSServiceException as __BaseException } from "./RDSServiceException";
export var AccountQuota;
(function (AccountQuota) {
    AccountQuota.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccountQuota || (AccountQuota = {}));
export var AccountAttributesMessage;
(function (AccountAttributesMessage) {
    AccountAttributesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccountAttributesMessage || (AccountAttributesMessage = {}));
export var AddRoleToDBClusterMessage;
(function (AddRoleToDBClusterMessage) {
    AddRoleToDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddRoleToDBClusterMessage || (AddRoleToDBClusterMessage = {}));
var DBClusterNotFoundFault = (function (_super) {
    __extends(DBClusterNotFoundFault, _super);
    function DBClusterNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterNotFoundFault.prototype);
        return _this;
    }
    return DBClusterNotFoundFault;
}(__BaseException));
export { DBClusterNotFoundFault };
var DBClusterRoleAlreadyExistsFault = (function (_super) {
    __extends(DBClusterRoleAlreadyExistsFault, _super);
    function DBClusterRoleAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterRoleAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterRoleAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterRoleAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterRoleAlreadyExistsFault;
}(__BaseException));
export { DBClusterRoleAlreadyExistsFault };
var DBClusterRoleQuotaExceededFault = (function (_super) {
    __extends(DBClusterRoleQuotaExceededFault, _super);
    function DBClusterRoleQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterRoleQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterRoleQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterRoleQuotaExceededFault.prototype);
        return _this;
    }
    return DBClusterRoleQuotaExceededFault;
}(__BaseException));
export { DBClusterRoleQuotaExceededFault };
var InvalidDBClusterStateFault = (function (_super) {
    __extends(InvalidDBClusterStateFault, _super);
    function InvalidDBClusterStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBClusterStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBClusterStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBClusterStateFault.prototype);
        return _this;
    }
    return InvalidDBClusterStateFault;
}(__BaseException));
export { InvalidDBClusterStateFault };
export var AddRoleToDBInstanceMessage;
(function (AddRoleToDBInstanceMessage) {
    AddRoleToDBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddRoleToDBInstanceMessage || (AddRoleToDBInstanceMessage = {}));
var DBInstanceNotFoundFault = (function (_super) {
    __extends(DBInstanceNotFoundFault, _super);
    function DBInstanceNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceNotFoundFault.prototype);
        return _this;
    }
    return DBInstanceNotFoundFault;
}(__BaseException));
export { DBInstanceNotFoundFault };
var DBInstanceRoleAlreadyExistsFault = (function (_super) {
    __extends(DBInstanceRoleAlreadyExistsFault, _super);
    function DBInstanceRoleAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceRoleAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceRoleAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceRoleAlreadyExistsFault.prototype);
        return _this;
    }
    return DBInstanceRoleAlreadyExistsFault;
}(__BaseException));
export { DBInstanceRoleAlreadyExistsFault };
var DBInstanceRoleQuotaExceededFault = (function (_super) {
    __extends(DBInstanceRoleQuotaExceededFault, _super);
    function DBInstanceRoleQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceRoleQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceRoleQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceRoleQuotaExceededFault.prototype);
        return _this;
    }
    return DBInstanceRoleQuotaExceededFault;
}(__BaseException));
export { DBInstanceRoleQuotaExceededFault };
var InvalidDBInstanceStateFault = (function (_super) {
    __extends(InvalidDBInstanceStateFault, _super);
    function InvalidDBInstanceStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBInstanceStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBInstanceStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBInstanceStateFault.prototype);
        return _this;
    }
    return InvalidDBInstanceStateFault;
}(__BaseException));
export { InvalidDBInstanceStateFault };
export var AddSourceIdentifierToSubscriptionMessage;
(function (AddSourceIdentifierToSubscriptionMessage) {
    AddSourceIdentifierToSubscriptionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddSourceIdentifierToSubscriptionMessage || (AddSourceIdentifierToSubscriptionMessage = {}));
export var EventSubscription;
(function (EventSubscription) {
    EventSubscription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EventSubscription || (EventSubscription = {}));
export var AddSourceIdentifierToSubscriptionResult;
(function (AddSourceIdentifierToSubscriptionResult) {
    AddSourceIdentifierToSubscriptionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddSourceIdentifierToSubscriptionResult || (AddSourceIdentifierToSubscriptionResult = {}));
var SourceNotFoundFault = (function (_super) {
    __extends(SourceNotFoundFault, _super);
    function SourceNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SourceNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SourceNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SourceNotFoundFault.prototype);
        return _this;
    }
    return SourceNotFoundFault;
}(__BaseException));
export { SourceNotFoundFault };
var SubscriptionNotFoundFault = (function (_super) {
    __extends(SubscriptionNotFoundFault, _super);
    function SubscriptionNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SubscriptionNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SubscriptionNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SubscriptionNotFoundFault.prototype);
        return _this;
    }
    return SubscriptionNotFoundFault;
}(__BaseException));
export { SubscriptionNotFoundFault };
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var AddTagsToResourceMessage;
(function (AddTagsToResourceMessage) {
    AddTagsToResourceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddTagsToResourceMessage || (AddTagsToResourceMessage = {}));
var DBProxyNotFoundFault = (function (_super) {
    __extends(DBProxyNotFoundFault, _super);
    function DBProxyNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyNotFoundFault.prototype);
        return _this;
    }
    return DBProxyNotFoundFault;
}(__BaseException));
export { DBProxyNotFoundFault };
var DBProxyTargetGroupNotFoundFault = (function (_super) {
    __extends(DBProxyTargetGroupNotFoundFault, _super);
    function DBProxyTargetGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyTargetGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyTargetGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyTargetGroupNotFoundFault.prototype);
        return _this;
    }
    return DBProxyTargetGroupNotFoundFault;
}(__BaseException));
export { DBProxyTargetGroupNotFoundFault };
var DBSnapshotNotFoundFault = (function (_super) {
    __extends(DBSnapshotNotFoundFault, _super);
    function DBSnapshotNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSnapshotNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSnapshotNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSnapshotNotFoundFault.prototype);
        return _this;
    }
    return DBSnapshotNotFoundFault;
}(__BaseException));
export { DBSnapshotNotFoundFault };
export var ApplyPendingMaintenanceActionMessage;
(function (ApplyPendingMaintenanceActionMessage) {
    ApplyPendingMaintenanceActionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ApplyPendingMaintenanceActionMessage || (ApplyPendingMaintenanceActionMessage = {}));
export var PendingMaintenanceAction;
(function (PendingMaintenanceAction) {
    PendingMaintenanceAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingMaintenanceAction || (PendingMaintenanceAction = {}));
export var ResourcePendingMaintenanceActions;
(function (ResourcePendingMaintenanceActions) {
    ResourcePendingMaintenanceActions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourcePendingMaintenanceActions || (ResourcePendingMaintenanceActions = {}));
export var ApplyPendingMaintenanceActionResult;
(function (ApplyPendingMaintenanceActionResult) {
    ApplyPendingMaintenanceActionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ApplyPendingMaintenanceActionResult || (ApplyPendingMaintenanceActionResult = {}));
var ResourceNotFoundFault = (function (_super) {
    __extends(ResourceNotFoundFault, _super);
    function ResourceNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "ResourceNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceNotFoundFault.prototype);
        return _this;
    }
    return ResourceNotFoundFault;
}(__BaseException));
export { ResourceNotFoundFault };
var AuthorizationAlreadyExistsFault = (function (_super) {
    __extends(AuthorizationAlreadyExistsFault, _super);
    function AuthorizationAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "AuthorizationAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "AuthorizationAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AuthorizationAlreadyExistsFault.prototype);
        return _this;
    }
    return AuthorizationAlreadyExistsFault;
}(__BaseException));
export { AuthorizationAlreadyExistsFault };
var AuthorizationQuotaExceededFault = (function (_super) {
    __extends(AuthorizationQuotaExceededFault, _super);
    function AuthorizationQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "AuthorizationQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "AuthorizationQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AuthorizationQuotaExceededFault.prototype);
        return _this;
    }
    return AuthorizationQuotaExceededFault;
}(__BaseException));
export { AuthorizationQuotaExceededFault };
export var AuthorizeDBSecurityGroupIngressMessage;
(function (AuthorizeDBSecurityGroupIngressMessage) {
    AuthorizeDBSecurityGroupIngressMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizeDBSecurityGroupIngressMessage || (AuthorizeDBSecurityGroupIngressMessage = {}));
export var EC2SecurityGroup;
(function (EC2SecurityGroup) {
    EC2SecurityGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EC2SecurityGroup || (EC2SecurityGroup = {}));
export var IPRange;
(function (IPRange) {
    IPRange.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IPRange || (IPRange = {}));
export var DBSecurityGroup;
(function (DBSecurityGroup) {
    DBSecurityGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroup || (DBSecurityGroup = {}));
export var AuthorizeDBSecurityGroupIngressResult;
(function (AuthorizeDBSecurityGroupIngressResult) {
    AuthorizeDBSecurityGroupIngressResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AuthorizeDBSecurityGroupIngressResult || (AuthorizeDBSecurityGroupIngressResult = {}));
var DBSecurityGroupNotFoundFault = (function (_super) {
    __extends(DBSecurityGroupNotFoundFault, _super);
    function DBSecurityGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSecurityGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSecurityGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSecurityGroupNotFoundFault.prototype);
        return _this;
    }
    return DBSecurityGroupNotFoundFault;
}(__BaseException));
export { DBSecurityGroupNotFoundFault };
var InvalidDBSecurityGroupStateFault = (function (_super) {
    __extends(InvalidDBSecurityGroupStateFault, _super);
    function InvalidDBSecurityGroupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSecurityGroupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSecurityGroupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSecurityGroupStateFault.prototype);
        return _this;
    }
    return InvalidDBSecurityGroupStateFault;
}(__BaseException));
export { InvalidDBSecurityGroupStateFault };
export var BacktrackDBClusterMessage;
(function (BacktrackDBClusterMessage) {
    BacktrackDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BacktrackDBClusterMessage || (BacktrackDBClusterMessage = {}));
export var DBClusterBacktrack;
(function (DBClusterBacktrack) {
    DBClusterBacktrack.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterBacktrack || (DBClusterBacktrack = {}));
export var CancelExportTaskMessage;
(function (CancelExportTaskMessage) {
    CancelExportTaskMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelExportTaskMessage || (CancelExportTaskMessage = {}));
export var ExportTask;
(function (ExportTask) {
    ExportTask.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportTask || (ExportTask = {}));
var ExportTaskNotFoundFault = (function (_super) {
    __extends(ExportTaskNotFoundFault, _super);
    function ExportTaskNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "ExportTaskNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "ExportTaskNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ExportTaskNotFoundFault.prototype);
        return _this;
    }
    return ExportTaskNotFoundFault;
}(__BaseException));
export { ExportTaskNotFoundFault };
var InvalidExportTaskStateFault = (function (_super) {
    __extends(InvalidExportTaskStateFault, _super);
    function InvalidExportTaskStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidExportTaskStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidExportTaskStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidExportTaskStateFault.prototype);
        return _this;
    }
    return InvalidExportTaskStateFault;
}(__BaseException));
export { InvalidExportTaskStateFault };
export var CopyDBClusterParameterGroupMessage;
(function (CopyDBClusterParameterGroupMessage) {
    CopyDBClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterParameterGroupMessage || (CopyDBClusterParameterGroupMessage = {}));
export var DBClusterParameterGroup;
(function (DBClusterParameterGroup) {
    DBClusterParameterGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterParameterGroup || (DBClusterParameterGroup = {}));
export var CopyDBClusterParameterGroupResult;
(function (CopyDBClusterParameterGroupResult) {
    CopyDBClusterParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterParameterGroupResult || (CopyDBClusterParameterGroupResult = {}));
var DBParameterGroupAlreadyExistsFault = (function (_super) {
    __extends(DBParameterGroupAlreadyExistsFault, _super);
    function DBParameterGroupAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBParameterGroupAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBParameterGroupAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBParameterGroupAlreadyExistsFault.prototype);
        return _this;
    }
    return DBParameterGroupAlreadyExistsFault;
}(__BaseException));
export { DBParameterGroupAlreadyExistsFault };
var DBParameterGroupNotFoundFault = (function (_super) {
    __extends(DBParameterGroupNotFoundFault, _super);
    function DBParameterGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBParameterGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBParameterGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBParameterGroupNotFoundFault.prototype);
        return _this;
    }
    return DBParameterGroupNotFoundFault;
}(__BaseException));
export { DBParameterGroupNotFoundFault };
var DBParameterGroupQuotaExceededFault = (function (_super) {
    __extends(DBParameterGroupQuotaExceededFault, _super);
    function DBParameterGroupQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBParameterGroupQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBParameterGroupQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBParameterGroupQuotaExceededFault.prototype);
        return _this;
    }
    return DBParameterGroupQuotaExceededFault;
}(__BaseException));
export { DBParameterGroupQuotaExceededFault };
export var CopyDBClusterSnapshotMessage;
(function (CopyDBClusterSnapshotMessage) {
    CopyDBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterSnapshotMessage || (CopyDBClusterSnapshotMessage = {}));
export var DBClusterSnapshot;
(function (DBClusterSnapshot) {
    DBClusterSnapshot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshot || (DBClusterSnapshot = {}));
export var CopyDBClusterSnapshotResult;
(function (CopyDBClusterSnapshotResult) {
    CopyDBClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBClusterSnapshotResult || (CopyDBClusterSnapshotResult = {}));
var DBClusterSnapshotAlreadyExistsFault = (function (_super) {
    __extends(DBClusterSnapshotAlreadyExistsFault, _super);
    function DBClusterSnapshotAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterSnapshotAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterSnapshotAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterSnapshotAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterSnapshotAlreadyExistsFault;
}(__BaseException));
export { DBClusterSnapshotAlreadyExistsFault };
var DBClusterSnapshotNotFoundFault = (function (_super) {
    __extends(DBClusterSnapshotNotFoundFault, _super);
    function DBClusterSnapshotNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterSnapshotNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterSnapshotNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterSnapshotNotFoundFault.prototype);
        return _this;
    }
    return DBClusterSnapshotNotFoundFault;
}(__BaseException));
export { DBClusterSnapshotNotFoundFault };
var InvalidDBClusterSnapshotStateFault = (function (_super) {
    __extends(InvalidDBClusterSnapshotStateFault, _super);
    function InvalidDBClusterSnapshotStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBClusterSnapshotStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBClusterSnapshotStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBClusterSnapshotStateFault.prototype);
        return _this;
    }
    return InvalidDBClusterSnapshotStateFault;
}(__BaseException));
export { InvalidDBClusterSnapshotStateFault };
var KMSKeyNotAccessibleFault = (function (_super) {
    __extends(KMSKeyNotAccessibleFault, _super);
    function KMSKeyNotAccessibleFault(opts) {
        var _this = _super.call(this, __assign({ name: "KMSKeyNotAccessibleFault", $fault: "client" }, opts)) || this;
        _this.name = "KMSKeyNotAccessibleFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, KMSKeyNotAccessibleFault.prototype);
        return _this;
    }
    return KMSKeyNotAccessibleFault;
}(__BaseException));
export { KMSKeyNotAccessibleFault };
var SnapshotQuotaExceededFault = (function (_super) {
    __extends(SnapshotQuotaExceededFault, _super);
    function SnapshotQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "SnapshotQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "SnapshotQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SnapshotQuotaExceededFault.prototype);
        return _this;
    }
    return SnapshotQuotaExceededFault;
}(__BaseException));
export { SnapshotQuotaExceededFault };
export var CopyDBParameterGroupMessage;
(function (CopyDBParameterGroupMessage) {
    CopyDBParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBParameterGroupMessage || (CopyDBParameterGroupMessage = {}));
export var DBParameterGroup;
(function (DBParameterGroup) {
    DBParameterGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroup || (DBParameterGroup = {}));
export var CopyDBParameterGroupResult;
(function (CopyDBParameterGroupResult) {
    CopyDBParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBParameterGroupResult || (CopyDBParameterGroupResult = {}));
export var CopyDBSnapshotMessage;
(function (CopyDBSnapshotMessage) {
    CopyDBSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBSnapshotMessage || (CopyDBSnapshotMessage = {}));
export var ProcessorFeature;
(function (ProcessorFeature) {
    ProcessorFeature.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProcessorFeature || (ProcessorFeature = {}));
export var DBSnapshot;
(function (DBSnapshot) {
    DBSnapshot.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSnapshot || (DBSnapshot = {}));
export var CopyDBSnapshotResult;
(function (CopyDBSnapshotResult) {
    CopyDBSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyDBSnapshotResult || (CopyDBSnapshotResult = {}));
var CustomAvailabilityZoneNotFoundFault = (function (_super) {
    __extends(CustomAvailabilityZoneNotFoundFault, _super);
    function CustomAvailabilityZoneNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "CustomAvailabilityZoneNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "CustomAvailabilityZoneNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomAvailabilityZoneNotFoundFault.prototype);
        return _this;
    }
    return CustomAvailabilityZoneNotFoundFault;
}(__BaseException));
export { CustomAvailabilityZoneNotFoundFault };
var DBSnapshotAlreadyExistsFault = (function (_super) {
    __extends(DBSnapshotAlreadyExistsFault, _super);
    function DBSnapshotAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSnapshotAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSnapshotAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSnapshotAlreadyExistsFault.prototype);
        return _this;
    }
    return DBSnapshotAlreadyExistsFault;
}(__BaseException));
export { DBSnapshotAlreadyExistsFault };
var InvalidDBSnapshotStateFault = (function (_super) {
    __extends(InvalidDBSnapshotStateFault, _super);
    function InvalidDBSnapshotStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSnapshotStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSnapshotStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSnapshotStateFault.prototype);
        return _this;
    }
    return InvalidDBSnapshotStateFault;
}(__BaseException));
export { InvalidDBSnapshotStateFault };
export var CopyOptionGroupMessage;
(function (CopyOptionGroupMessage) {
    CopyOptionGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyOptionGroupMessage || (CopyOptionGroupMessage = {}));
export var DBSecurityGroupMembership;
(function (DBSecurityGroupMembership) {
    DBSecurityGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSecurityGroupMembership || (DBSecurityGroupMembership = {}));
export var OptionSetting;
(function (OptionSetting) {
    OptionSetting.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionSetting || (OptionSetting = {}));
export var VpcSecurityGroupMembership;
(function (VpcSecurityGroupMembership) {
    VpcSecurityGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VpcSecurityGroupMembership || (VpcSecurityGroupMembership = {}));
export var Option;
(function (Option) {
    Option.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Option || (Option = {}));
export var OptionGroup;
(function (OptionGroup) {
    OptionGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionGroup || (OptionGroup = {}));
export var CopyOptionGroupResult;
(function (CopyOptionGroupResult) {
    CopyOptionGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CopyOptionGroupResult || (CopyOptionGroupResult = {}));
var OptionGroupAlreadyExistsFault = (function (_super) {
    __extends(OptionGroupAlreadyExistsFault, _super);
    function OptionGroupAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "OptionGroupAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "OptionGroupAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, OptionGroupAlreadyExistsFault.prototype);
        return _this;
    }
    return OptionGroupAlreadyExistsFault;
}(__BaseException));
export { OptionGroupAlreadyExistsFault };
var OptionGroupNotFoundFault = (function (_super) {
    __extends(OptionGroupNotFoundFault, _super);
    function OptionGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "OptionGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "OptionGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, OptionGroupNotFoundFault.prototype);
        return _this;
    }
    return OptionGroupNotFoundFault;
}(__BaseException));
export { OptionGroupNotFoundFault };
var OptionGroupQuotaExceededFault = (function (_super) {
    __extends(OptionGroupQuotaExceededFault, _super);
    function OptionGroupQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "OptionGroupQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "OptionGroupQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, OptionGroupQuotaExceededFault.prototype);
        return _this;
    }
    return OptionGroupQuotaExceededFault;
}(__BaseException));
export { OptionGroupQuotaExceededFault };
export var CreateCustomAvailabilityZoneMessage;
(function (CreateCustomAvailabilityZoneMessage) {
    CreateCustomAvailabilityZoneMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCustomAvailabilityZoneMessage || (CreateCustomAvailabilityZoneMessage = {}));
export var VpnDetails;
(function (VpnDetails) {
    VpnDetails.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.VpnPSK && { VpnPSK: SENSITIVE_STRING }))); };
})(VpnDetails || (VpnDetails = {}));
export var CustomAvailabilityZone;
(function (CustomAvailabilityZone) {
    CustomAvailabilityZone.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.VpnDetails && { VpnDetails: VpnDetails.filterSensitiveLog(obj.VpnDetails) }))); };
})(CustomAvailabilityZone || (CustomAvailabilityZone = {}));
export var CreateCustomAvailabilityZoneResult;
(function (CreateCustomAvailabilityZoneResult) {
    CreateCustomAvailabilityZoneResult.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.CustomAvailabilityZone && {
        CustomAvailabilityZone: CustomAvailabilityZone.filterSensitiveLog(obj.CustomAvailabilityZone),
    }))); };
})(CreateCustomAvailabilityZoneResult || (CreateCustomAvailabilityZoneResult = {}));
var CustomAvailabilityZoneAlreadyExistsFault = (function (_super) {
    __extends(CustomAvailabilityZoneAlreadyExistsFault, _super);
    function CustomAvailabilityZoneAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "CustomAvailabilityZoneAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "CustomAvailabilityZoneAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomAvailabilityZoneAlreadyExistsFault.prototype);
        return _this;
    }
    return CustomAvailabilityZoneAlreadyExistsFault;
}(__BaseException));
export { CustomAvailabilityZoneAlreadyExistsFault };
var CustomAvailabilityZoneQuotaExceededFault = (function (_super) {
    __extends(CustomAvailabilityZoneQuotaExceededFault, _super);
    function CustomAvailabilityZoneQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "CustomAvailabilityZoneQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "CustomAvailabilityZoneQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomAvailabilityZoneQuotaExceededFault.prototype);
        return _this;
    }
    return CustomAvailabilityZoneQuotaExceededFault;
}(__BaseException));
export { CustomAvailabilityZoneQuotaExceededFault };
export var CreateCustomDBEngineVersionMessage;
(function (CreateCustomDBEngineVersionMessage) {
    CreateCustomDBEngineVersionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCustomDBEngineVersionMessage || (CreateCustomDBEngineVersionMessage = {}));
var CustomDBEngineVersionAlreadyExistsFault = (function (_super) {
    __extends(CustomDBEngineVersionAlreadyExistsFault, _super);
    function CustomDBEngineVersionAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "CustomDBEngineVersionAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "CustomDBEngineVersionAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomDBEngineVersionAlreadyExistsFault.prototype);
        return _this;
    }
    return CustomDBEngineVersionAlreadyExistsFault;
}(__BaseException));
export { CustomDBEngineVersionAlreadyExistsFault };
var CustomDBEngineVersionQuotaExceededFault = (function (_super) {
    __extends(CustomDBEngineVersionQuotaExceededFault, _super);
    function CustomDBEngineVersionQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "CustomDBEngineVersionQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "CustomDBEngineVersionQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomDBEngineVersionQuotaExceededFault.prototype);
        return _this;
    }
    return CustomDBEngineVersionQuotaExceededFault;
}(__BaseException));
export { CustomDBEngineVersionQuotaExceededFault };
export var CharacterSet;
(function (CharacterSet) {
    CharacterSet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CharacterSet || (CharacterSet = {}));
export var Timezone;
(function (Timezone) {
    Timezone.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Timezone || (Timezone = {}));
export var UpgradeTarget;
(function (UpgradeTarget) {
    UpgradeTarget.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpgradeTarget || (UpgradeTarget = {}));
export var DBEngineVersion;
(function (DBEngineVersion) {
    DBEngineVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBEngineVersion || (DBEngineVersion = {}));
export var ScalingConfiguration;
(function (ScalingConfiguration) {
    ScalingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScalingConfiguration || (ScalingConfiguration = {}));
export var CreateDBClusterMessage;
(function (CreateDBClusterMessage) {
    CreateDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterMessage || (CreateDBClusterMessage = {}));
export var DBClusterRole;
(function (DBClusterRole) {
    DBClusterRole.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterRole || (DBClusterRole = {}));
export var DBClusterMember;
(function (DBClusterMember) {
    DBClusterMember.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterMember || (DBClusterMember = {}));
export var DBClusterOptionGroupStatus;
(function (DBClusterOptionGroupStatus) {
    DBClusterOptionGroupStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterOptionGroupStatus || (DBClusterOptionGroupStatus = {}));
export var DomainMembership;
(function (DomainMembership) {
    DomainMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainMembership || (DomainMembership = {}));
export var WriteForwardingStatus;
(function (WriteForwardingStatus) {
    WriteForwardingStatus["DISABLED"] = "disabled";
    WriteForwardingStatus["DISABLING"] = "disabling";
    WriteForwardingStatus["ENABLED"] = "enabled";
    WriteForwardingStatus["ENABLING"] = "enabling";
    WriteForwardingStatus["UNKNOWN"] = "unknown";
})(WriteForwardingStatus || (WriteForwardingStatus = {}));
export var PendingCloudwatchLogsExports;
(function (PendingCloudwatchLogsExports) {
    PendingCloudwatchLogsExports.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingCloudwatchLogsExports || (PendingCloudwatchLogsExports = {}));
export var ClusterPendingModifiedValues;
(function (ClusterPendingModifiedValues) {
    ClusterPendingModifiedValues.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ClusterPendingModifiedValues || (ClusterPendingModifiedValues = {}));
export var ScalingConfigurationInfo;
(function (ScalingConfigurationInfo) {
    ScalingConfigurationInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScalingConfigurationInfo || (ScalingConfigurationInfo = {}));
export var DBCluster;
(function (DBCluster) {
    DBCluster.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBCluster || (DBCluster = {}));
export var CreateDBClusterResult;
(function (CreateDBClusterResult) {
    CreateDBClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterResult || (CreateDBClusterResult = {}));
var DBClusterAlreadyExistsFault = (function (_super) {
    __extends(DBClusterAlreadyExistsFault, _super);
    function DBClusterAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterAlreadyExistsFault;
}(__BaseException));
export { DBClusterAlreadyExistsFault };
var DBClusterParameterGroupNotFoundFault = (function (_super) {
    __extends(DBClusterParameterGroupNotFoundFault, _super);
    function DBClusterParameterGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterParameterGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterParameterGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterParameterGroupNotFoundFault.prototype);
        return _this;
    }
    return DBClusterParameterGroupNotFoundFault;
}(__BaseException));
export { DBClusterParameterGroupNotFoundFault };
var DBClusterQuotaExceededFault = (function (_super) {
    __extends(DBClusterQuotaExceededFault, _super);
    function DBClusterQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterQuotaExceededFault.prototype);
        return _this;
    }
    return DBClusterQuotaExceededFault;
}(__BaseException));
export { DBClusterQuotaExceededFault };
var DBSubnetGroupDoesNotCoverEnoughAZs = (function (_super) {
    __extends(DBSubnetGroupDoesNotCoverEnoughAZs, _super);
    function DBSubnetGroupDoesNotCoverEnoughAZs(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupDoesNotCoverEnoughAZs", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupDoesNotCoverEnoughAZs";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupDoesNotCoverEnoughAZs.prototype);
        return _this;
    }
    return DBSubnetGroupDoesNotCoverEnoughAZs;
}(__BaseException));
export { DBSubnetGroupDoesNotCoverEnoughAZs };
var DBSubnetGroupNotFoundFault = (function (_super) {
    __extends(DBSubnetGroupNotFoundFault, _super);
    function DBSubnetGroupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupNotFoundFault.prototype);
        return _this;
    }
    return DBSubnetGroupNotFoundFault;
}(__BaseException));
export { DBSubnetGroupNotFoundFault };
var DomainNotFoundFault = (function (_super) {
    __extends(DomainNotFoundFault, _super);
    function DomainNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DomainNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DomainNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DomainNotFoundFault.prototype);
        return _this;
    }
    return DomainNotFoundFault;
}(__BaseException));
export { DomainNotFoundFault };
var GlobalClusterNotFoundFault = (function (_super) {
    __extends(GlobalClusterNotFoundFault, _super);
    function GlobalClusterNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "GlobalClusterNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "GlobalClusterNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, GlobalClusterNotFoundFault.prototype);
        return _this;
    }
    return GlobalClusterNotFoundFault;
}(__BaseException));
export { GlobalClusterNotFoundFault };
var InsufficientStorageClusterCapacityFault = (function (_super) {
    __extends(InsufficientStorageClusterCapacityFault, _super);
    function InsufficientStorageClusterCapacityFault(opts) {
        var _this = _super.call(this, __assign({ name: "InsufficientStorageClusterCapacityFault", $fault: "client" }, opts)) || this;
        _this.name = "InsufficientStorageClusterCapacityFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InsufficientStorageClusterCapacityFault.prototype);
        return _this;
    }
    return InsufficientStorageClusterCapacityFault;
}(__BaseException));
export { InsufficientStorageClusterCapacityFault };
var InvalidDBSubnetGroupStateFault = (function (_super) {
    __extends(InvalidDBSubnetGroupStateFault, _super);
    function InvalidDBSubnetGroupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSubnetGroupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSubnetGroupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSubnetGroupStateFault.prototype);
        return _this;
    }
    return InvalidDBSubnetGroupStateFault;
}(__BaseException));
export { InvalidDBSubnetGroupStateFault };
var InvalidGlobalClusterStateFault = (function (_super) {
    __extends(InvalidGlobalClusterStateFault, _super);
    function InvalidGlobalClusterStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidGlobalClusterStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidGlobalClusterStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidGlobalClusterStateFault.prototype);
        return _this;
    }
    return InvalidGlobalClusterStateFault;
}(__BaseException));
export { InvalidGlobalClusterStateFault };
var InvalidSubnet = (function (_super) {
    __extends(InvalidSubnet, _super);
    function InvalidSubnet(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidSubnet", $fault: "client" }, opts)) || this;
        _this.name = "InvalidSubnet";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidSubnet.prototype);
        return _this;
    }
    return InvalidSubnet;
}(__BaseException));
export { InvalidSubnet };
var InvalidVPCNetworkStateFault = (function (_super) {
    __extends(InvalidVPCNetworkStateFault, _super);
    function InvalidVPCNetworkStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidVPCNetworkStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidVPCNetworkStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidVPCNetworkStateFault.prototype);
        return _this;
    }
    return InvalidVPCNetworkStateFault;
}(__BaseException));
export { InvalidVPCNetworkStateFault };
var StorageQuotaExceededFault = (function (_super) {
    __extends(StorageQuotaExceededFault, _super);
    function StorageQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "StorageQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "StorageQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, StorageQuotaExceededFault.prototype);
        return _this;
    }
    return StorageQuotaExceededFault;
}(__BaseException));
export { StorageQuotaExceededFault };
export var CreateDBClusterEndpointMessage;
(function (CreateDBClusterEndpointMessage) {
    CreateDBClusterEndpointMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterEndpointMessage || (CreateDBClusterEndpointMessage = {}));
export var DBClusterEndpoint;
(function (DBClusterEndpoint) {
    DBClusterEndpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterEndpoint || (DBClusterEndpoint = {}));
var DBClusterEndpointAlreadyExistsFault = (function (_super) {
    __extends(DBClusterEndpointAlreadyExistsFault, _super);
    function DBClusterEndpointAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterEndpointAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterEndpointAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterEndpointAlreadyExistsFault.prototype);
        return _this;
    }
    return DBClusterEndpointAlreadyExistsFault;
}(__BaseException));
export { DBClusterEndpointAlreadyExistsFault };
var DBClusterEndpointQuotaExceededFault = (function (_super) {
    __extends(DBClusterEndpointQuotaExceededFault, _super);
    function DBClusterEndpointQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterEndpointQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterEndpointQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterEndpointQuotaExceededFault.prototype);
        return _this;
    }
    return DBClusterEndpointQuotaExceededFault;
}(__BaseException));
export { DBClusterEndpointQuotaExceededFault };
export var CreateDBClusterParameterGroupMessage;
(function (CreateDBClusterParameterGroupMessage) {
    CreateDBClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterParameterGroupMessage || (CreateDBClusterParameterGroupMessage = {}));
export var CreateDBClusterParameterGroupResult;
(function (CreateDBClusterParameterGroupResult) {
    CreateDBClusterParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterParameterGroupResult || (CreateDBClusterParameterGroupResult = {}));
export var CreateDBClusterSnapshotMessage;
(function (CreateDBClusterSnapshotMessage) {
    CreateDBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterSnapshotMessage || (CreateDBClusterSnapshotMessage = {}));
export var CreateDBClusterSnapshotResult;
(function (CreateDBClusterSnapshotResult) {
    CreateDBClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBClusterSnapshotResult || (CreateDBClusterSnapshotResult = {}));
var AuthorizationNotFoundFault = (function (_super) {
    __extends(AuthorizationNotFoundFault, _super);
    function AuthorizationNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "AuthorizationNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "AuthorizationNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AuthorizationNotFoundFault.prototype);
        return _this;
    }
    return AuthorizationNotFoundFault;
}(__BaseException));
export { AuthorizationNotFoundFault };
var BackupPolicyNotFoundFault = (function (_super) {
    __extends(BackupPolicyNotFoundFault, _super);
    function BackupPolicyNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "BackupPolicyNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "BackupPolicyNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, BackupPolicyNotFoundFault.prototype);
        return _this;
    }
    return BackupPolicyNotFoundFault;
}(__BaseException));
export { BackupPolicyNotFoundFault };
export var CreateDBInstanceMessage;
(function (CreateDBInstanceMessage) {
    CreateDBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceMessage || (CreateDBInstanceMessage = {}));
export var DBInstanceRole;
(function (DBInstanceRole) {
    DBInstanceRole.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceRole || (DBInstanceRole = {}));
export var AutomationMode;
(function (AutomationMode) {
    AutomationMode["ALL_PAUSED"] = "all-paused";
    AutomationMode["FULL"] = "full";
})(AutomationMode || (AutomationMode = {}));
export var DBInstanceAutomatedBackupsReplication;
(function (DBInstanceAutomatedBackupsReplication) {
    DBInstanceAutomatedBackupsReplication.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackupsReplication || (DBInstanceAutomatedBackupsReplication = {}));
export var DBParameterGroupStatus;
(function (DBParameterGroupStatus) {
    DBParameterGroupStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupStatus || (DBParameterGroupStatus = {}));
export var AvailabilityZone;
(function (AvailabilityZone) {
    AvailabilityZone.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AvailabilityZone || (AvailabilityZone = {}));
export var Outpost;
(function (Outpost) {
    Outpost.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Outpost || (Outpost = {}));
export var Subnet;
(function (Subnet) {
    Subnet.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Subnet || (Subnet = {}));
export var DBSubnetGroup;
(function (DBSubnetGroup) {
    DBSubnetGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBSubnetGroup || (DBSubnetGroup = {}));
export var Endpoint;
(function (Endpoint) {
    Endpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Endpoint || (Endpoint = {}));
export var OptionGroupMembership;
(function (OptionGroupMembership) {
    OptionGroupMembership.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionGroupMembership || (OptionGroupMembership = {}));
export var PendingModifiedValues;
(function (PendingModifiedValues) {
    PendingModifiedValues.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PendingModifiedValues || (PendingModifiedValues = {}));
export var ReplicaMode;
(function (ReplicaMode) {
    ReplicaMode["MOUNTED"] = "mounted";
    ReplicaMode["OPEN_READ_ONLY"] = "open-read-only";
})(ReplicaMode || (ReplicaMode = {}));
export var DBInstanceStatusInfo;
(function (DBInstanceStatusInfo) {
    DBInstanceStatusInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceStatusInfo || (DBInstanceStatusInfo = {}));
export var DBInstance;
(function (DBInstance) {
    DBInstance.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstance || (DBInstance = {}));
export var CreateDBInstanceResult;
(function (CreateDBInstanceResult) {
    CreateDBInstanceResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceResult || (CreateDBInstanceResult = {}));
var DBInstanceAlreadyExistsFault = (function (_super) {
    __extends(DBInstanceAlreadyExistsFault, _super);
    function DBInstanceAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceAlreadyExistsFault.prototype);
        return _this;
    }
    return DBInstanceAlreadyExistsFault;
}(__BaseException));
export { DBInstanceAlreadyExistsFault };
var InstanceQuotaExceededFault = (function (_super) {
    __extends(InstanceQuotaExceededFault, _super);
    function InstanceQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "InstanceQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "InstanceQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InstanceQuotaExceededFault.prototype);
        return _this;
    }
    return InstanceQuotaExceededFault;
}(__BaseException));
export { InstanceQuotaExceededFault };
var InsufficientDBInstanceCapacityFault = (function (_super) {
    __extends(InsufficientDBInstanceCapacityFault, _super);
    function InsufficientDBInstanceCapacityFault(opts) {
        var _this = _super.call(this, __assign({ name: "InsufficientDBInstanceCapacityFault", $fault: "client" }, opts)) || this;
        _this.name = "InsufficientDBInstanceCapacityFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InsufficientDBInstanceCapacityFault.prototype);
        return _this;
    }
    return InsufficientDBInstanceCapacityFault;
}(__BaseException));
export { InsufficientDBInstanceCapacityFault };
var ProvisionedIopsNotAvailableInAZFault = (function (_super) {
    __extends(ProvisionedIopsNotAvailableInAZFault, _super);
    function ProvisionedIopsNotAvailableInAZFault(opts) {
        var _this = _super.call(this, __assign({ name: "ProvisionedIopsNotAvailableInAZFault", $fault: "client" }, opts)) || this;
        _this.name = "ProvisionedIopsNotAvailableInAZFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ProvisionedIopsNotAvailableInAZFault.prototype);
        return _this;
    }
    return ProvisionedIopsNotAvailableInAZFault;
}(__BaseException));
export { ProvisionedIopsNotAvailableInAZFault };
var StorageTypeNotSupportedFault = (function (_super) {
    __extends(StorageTypeNotSupportedFault, _super);
    function StorageTypeNotSupportedFault(opts) {
        var _this = _super.call(this, __assign({ name: "StorageTypeNotSupportedFault", $fault: "client" }, opts)) || this;
        _this.name = "StorageTypeNotSupportedFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, StorageTypeNotSupportedFault.prototype);
        return _this;
    }
    return StorageTypeNotSupportedFault;
}(__BaseException));
export { StorageTypeNotSupportedFault };
export var CreateDBInstanceReadReplicaMessage;
(function (CreateDBInstanceReadReplicaMessage) {
    CreateDBInstanceReadReplicaMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceReadReplicaMessage || (CreateDBInstanceReadReplicaMessage = {}));
export var CreateDBInstanceReadReplicaResult;
(function (CreateDBInstanceReadReplicaResult) {
    CreateDBInstanceReadReplicaResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBInstanceReadReplicaResult || (CreateDBInstanceReadReplicaResult = {}));
var DBSubnetGroupNotAllowedFault = (function (_super) {
    __extends(DBSubnetGroupNotAllowedFault, _super);
    function DBSubnetGroupNotAllowedFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupNotAllowedFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupNotAllowedFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupNotAllowedFault.prototype);
        return _this;
    }
    return DBSubnetGroupNotAllowedFault;
}(__BaseException));
export { DBSubnetGroupNotAllowedFault };
var InvalidDBSubnetGroupFault = (function (_super) {
    __extends(InvalidDBSubnetGroupFault, _super);
    function InvalidDBSubnetGroupFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSubnetGroupFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSubnetGroupFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSubnetGroupFault.prototype);
        return _this;
    }
    return InvalidDBSubnetGroupFault;
}(__BaseException));
export { InvalidDBSubnetGroupFault };
export var CreateDBParameterGroupMessage;
(function (CreateDBParameterGroupMessage) {
    CreateDBParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBParameterGroupMessage || (CreateDBParameterGroupMessage = {}));
export var CreateDBParameterGroupResult;
(function (CreateDBParameterGroupResult) {
    CreateDBParameterGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBParameterGroupResult || (CreateDBParameterGroupResult = {}));
export var AuthScheme;
(function (AuthScheme) {
    AuthScheme["SECRETS"] = "SECRETS";
})(AuthScheme || (AuthScheme = {}));
export var IAMAuthMode;
(function (IAMAuthMode) {
    IAMAuthMode["DISABLED"] = "DISABLED";
    IAMAuthMode["REQUIRED"] = "REQUIRED";
})(IAMAuthMode || (IAMAuthMode = {}));
export var UserAuthConfig;
(function (UserAuthConfig) {
    UserAuthConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserAuthConfig || (UserAuthConfig = {}));
export var EngineFamily;
(function (EngineFamily) {
    EngineFamily["MYSQL"] = "MYSQL";
    EngineFamily["POSTGRESQL"] = "POSTGRESQL";
})(EngineFamily || (EngineFamily = {}));
export var CreateDBProxyRequest;
(function (CreateDBProxyRequest) {
    CreateDBProxyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyRequest || (CreateDBProxyRequest = {}));
export var UserAuthConfigInfo;
(function (UserAuthConfigInfo) {
    UserAuthConfigInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UserAuthConfigInfo || (UserAuthConfigInfo = {}));
export var DBProxyStatus;
(function (DBProxyStatus) {
    DBProxyStatus["AVAILABLE"] = "available";
    DBProxyStatus["CREATING"] = "creating";
    DBProxyStatus["DELETING"] = "deleting";
    DBProxyStatus["INCOMPATIBLE_NETWORK"] = "incompatible-network";
    DBProxyStatus["INSUFFICIENT_RESOURCE_LIMITS"] = "insufficient-resource-limits";
    DBProxyStatus["MODIFYING"] = "modifying";
    DBProxyStatus["REACTIVATING"] = "reactivating";
    DBProxyStatus["SUSPENDED"] = "suspended";
    DBProxyStatus["SUSPENDING"] = "suspending";
})(DBProxyStatus || (DBProxyStatus = {}));
export var DBProxy;
(function (DBProxy) {
    DBProxy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxy || (DBProxy = {}));
export var CreateDBProxyResponse;
(function (CreateDBProxyResponse) {
    CreateDBProxyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyResponse || (CreateDBProxyResponse = {}));
var DBProxyAlreadyExistsFault = (function (_super) {
    __extends(DBProxyAlreadyExistsFault, _super);
    function DBProxyAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyAlreadyExistsFault.prototype);
        return _this;
    }
    return DBProxyAlreadyExistsFault;
}(__BaseException));
export { DBProxyAlreadyExistsFault };
var DBProxyQuotaExceededFault = (function (_super) {
    __extends(DBProxyQuotaExceededFault, _super);
    function DBProxyQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyQuotaExceededFault.prototype);
        return _this;
    }
    return DBProxyQuotaExceededFault;
}(__BaseException));
export { DBProxyQuotaExceededFault };
export var DBProxyEndpointTargetRole;
(function (DBProxyEndpointTargetRole) {
    DBProxyEndpointTargetRole["READ_ONLY"] = "READ_ONLY";
    DBProxyEndpointTargetRole["READ_WRITE"] = "READ_WRITE";
})(DBProxyEndpointTargetRole || (DBProxyEndpointTargetRole = {}));
export var CreateDBProxyEndpointRequest;
(function (CreateDBProxyEndpointRequest) {
    CreateDBProxyEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyEndpointRequest || (CreateDBProxyEndpointRequest = {}));
export var DBProxyEndpointStatus;
(function (DBProxyEndpointStatus) {
    DBProxyEndpointStatus["AVAILABLE"] = "available";
    DBProxyEndpointStatus["CREATING"] = "creating";
    DBProxyEndpointStatus["DELETING"] = "deleting";
    DBProxyEndpointStatus["INCOMPATIBLE_NETWORK"] = "incompatible-network";
    DBProxyEndpointStatus["INSUFFICIENT_RESOURCE_LIMITS"] = "insufficient-resource-limits";
    DBProxyEndpointStatus["MODIFYING"] = "modifying";
})(DBProxyEndpointStatus || (DBProxyEndpointStatus = {}));
export var DBProxyEndpoint;
(function (DBProxyEndpoint) {
    DBProxyEndpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyEndpoint || (DBProxyEndpoint = {}));
export var CreateDBProxyEndpointResponse;
(function (CreateDBProxyEndpointResponse) {
    CreateDBProxyEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBProxyEndpointResponse || (CreateDBProxyEndpointResponse = {}));
var DBProxyEndpointAlreadyExistsFault = (function (_super) {
    __extends(DBProxyEndpointAlreadyExistsFault, _super);
    function DBProxyEndpointAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyEndpointAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyEndpointAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyEndpointAlreadyExistsFault.prototype);
        return _this;
    }
    return DBProxyEndpointAlreadyExistsFault;
}(__BaseException));
export { DBProxyEndpointAlreadyExistsFault };
var DBProxyEndpointQuotaExceededFault = (function (_super) {
    __extends(DBProxyEndpointQuotaExceededFault, _super);
    function DBProxyEndpointQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyEndpointQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyEndpointQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyEndpointQuotaExceededFault.prototype);
        return _this;
    }
    return DBProxyEndpointQuotaExceededFault;
}(__BaseException));
export { DBProxyEndpointQuotaExceededFault };
var InvalidDBProxyStateFault = (function (_super) {
    __extends(InvalidDBProxyStateFault, _super);
    function InvalidDBProxyStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBProxyStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBProxyStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBProxyStateFault.prototype);
        return _this;
    }
    return InvalidDBProxyStateFault;
}(__BaseException));
export { InvalidDBProxyStateFault };
export var CreateDBSecurityGroupMessage;
(function (CreateDBSecurityGroupMessage) {
    CreateDBSecurityGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSecurityGroupMessage || (CreateDBSecurityGroupMessage = {}));
export var CreateDBSecurityGroupResult;
(function (CreateDBSecurityGroupResult) {
    CreateDBSecurityGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSecurityGroupResult || (CreateDBSecurityGroupResult = {}));
var DBSecurityGroupAlreadyExistsFault = (function (_super) {
    __extends(DBSecurityGroupAlreadyExistsFault, _super);
    function DBSecurityGroupAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSecurityGroupAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSecurityGroupAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSecurityGroupAlreadyExistsFault.prototype);
        return _this;
    }
    return DBSecurityGroupAlreadyExistsFault;
}(__BaseException));
export { DBSecurityGroupAlreadyExistsFault };
var DBSecurityGroupNotSupportedFault = (function (_super) {
    __extends(DBSecurityGroupNotSupportedFault, _super);
    function DBSecurityGroupNotSupportedFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSecurityGroupNotSupportedFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSecurityGroupNotSupportedFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSecurityGroupNotSupportedFault.prototype);
        return _this;
    }
    return DBSecurityGroupNotSupportedFault;
}(__BaseException));
export { DBSecurityGroupNotSupportedFault };
var DBSecurityGroupQuotaExceededFault = (function (_super) {
    __extends(DBSecurityGroupQuotaExceededFault, _super);
    function DBSecurityGroupQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSecurityGroupQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSecurityGroupQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSecurityGroupQuotaExceededFault.prototype);
        return _this;
    }
    return DBSecurityGroupQuotaExceededFault;
}(__BaseException));
export { DBSecurityGroupQuotaExceededFault };
export var CreateDBSnapshotMessage;
(function (CreateDBSnapshotMessage) {
    CreateDBSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSnapshotMessage || (CreateDBSnapshotMessage = {}));
export var CreateDBSnapshotResult;
(function (CreateDBSnapshotResult) {
    CreateDBSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSnapshotResult || (CreateDBSnapshotResult = {}));
export var CreateDBSubnetGroupMessage;
(function (CreateDBSubnetGroupMessage) {
    CreateDBSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSubnetGroupMessage || (CreateDBSubnetGroupMessage = {}));
export var CreateDBSubnetGroupResult;
(function (CreateDBSubnetGroupResult) {
    CreateDBSubnetGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDBSubnetGroupResult || (CreateDBSubnetGroupResult = {}));
var DBSubnetGroupAlreadyExistsFault = (function (_super) {
    __extends(DBSubnetGroupAlreadyExistsFault, _super);
    function DBSubnetGroupAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupAlreadyExistsFault.prototype);
        return _this;
    }
    return DBSubnetGroupAlreadyExistsFault;
}(__BaseException));
export { DBSubnetGroupAlreadyExistsFault };
var DBSubnetGroupQuotaExceededFault = (function (_super) {
    __extends(DBSubnetGroupQuotaExceededFault, _super);
    function DBSubnetGroupQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetGroupQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetGroupQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetGroupQuotaExceededFault.prototype);
        return _this;
    }
    return DBSubnetGroupQuotaExceededFault;
}(__BaseException));
export { DBSubnetGroupQuotaExceededFault };
var DBSubnetQuotaExceededFault = (function (_super) {
    __extends(DBSubnetQuotaExceededFault, _super);
    function DBSubnetQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBSubnetQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBSubnetQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBSubnetQuotaExceededFault.prototype);
        return _this;
    }
    return DBSubnetQuotaExceededFault;
}(__BaseException));
export { DBSubnetQuotaExceededFault };
export var CreateEventSubscriptionMessage;
(function (CreateEventSubscriptionMessage) {
    CreateEventSubscriptionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEventSubscriptionMessage || (CreateEventSubscriptionMessage = {}));
export var CreateEventSubscriptionResult;
(function (CreateEventSubscriptionResult) {
    CreateEventSubscriptionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEventSubscriptionResult || (CreateEventSubscriptionResult = {}));
var EventSubscriptionQuotaExceededFault = (function (_super) {
    __extends(EventSubscriptionQuotaExceededFault, _super);
    function EventSubscriptionQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "EventSubscriptionQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "EventSubscriptionQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, EventSubscriptionQuotaExceededFault.prototype);
        return _this;
    }
    return EventSubscriptionQuotaExceededFault;
}(__BaseException));
export { EventSubscriptionQuotaExceededFault };
var SNSInvalidTopicFault = (function (_super) {
    __extends(SNSInvalidTopicFault, _super);
    function SNSInvalidTopicFault(opts) {
        var _this = _super.call(this, __assign({ name: "SNSInvalidTopicFault", $fault: "client" }, opts)) || this;
        _this.name = "SNSInvalidTopicFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SNSInvalidTopicFault.prototype);
        return _this;
    }
    return SNSInvalidTopicFault;
}(__BaseException));
export { SNSInvalidTopicFault };
var SNSNoAuthorizationFault = (function (_super) {
    __extends(SNSNoAuthorizationFault, _super);
    function SNSNoAuthorizationFault(opts) {
        var _this = _super.call(this, __assign({ name: "SNSNoAuthorizationFault", $fault: "client" }, opts)) || this;
        _this.name = "SNSNoAuthorizationFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SNSNoAuthorizationFault.prototype);
        return _this;
    }
    return SNSNoAuthorizationFault;
}(__BaseException));
export { SNSNoAuthorizationFault };
var SNSTopicArnNotFoundFault = (function (_super) {
    __extends(SNSTopicArnNotFoundFault, _super);
    function SNSTopicArnNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SNSTopicArnNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SNSTopicArnNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SNSTopicArnNotFoundFault.prototype);
        return _this;
    }
    return SNSTopicArnNotFoundFault;
}(__BaseException));
export { SNSTopicArnNotFoundFault };
var SubscriptionAlreadyExistFault = (function (_super) {
    __extends(SubscriptionAlreadyExistFault, _super);
    function SubscriptionAlreadyExistFault(opts) {
        var _this = _super.call(this, __assign({ name: "SubscriptionAlreadyExistFault", $fault: "client" }, opts)) || this;
        _this.name = "SubscriptionAlreadyExistFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SubscriptionAlreadyExistFault.prototype);
        return _this;
    }
    return SubscriptionAlreadyExistFault;
}(__BaseException));
export { SubscriptionAlreadyExistFault };
var SubscriptionCategoryNotFoundFault = (function (_super) {
    __extends(SubscriptionCategoryNotFoundFault, _super);
    function SubscriptionCategoryNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "SubscriptionCategoryNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "SubscriptionCategoryNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SubscriptionCategoryNotFoundFault.prototype);
        return _this;
    }
    return SubscriptionCategoryNotFoundFault;
}(__BaseException));
export { SubscriptionCategoryNotFoundFault };
export var CreateGlobalClusterMessage;
(function (CreateGlobalClusterMessage) {
    CreateGlobalClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalClusterMessage || (CreateGlobalClusterMessage = {}));
export var FailoverStatus;
(function (FailoverStatus) {
    FailoverStatus["CANCELLING"] = "cancelling";
    FailoverStatus["FAILING_OVER"] = "failing-over";
    FailoverStatus["PENDING"] = "pending";
})(FailoverStatus || (FailoverStatus = {}));
export var FailoverState;
(function (FailoverState) {
    FailoverState.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailoverState || (FailoverState = {}));
export var GlobalClusterMember;
(function (GlobalClusterMember) {
    GlobalClusterMember.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalClusterMember || (GlobalClusterMember = {}));
export var GlobalCluster;
(function (GlobalCluster) {
    GlobalCluster.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalCluster || (GlobalCluster = {}));
export var CreateGlobalClusterResult;
(function (CreateGlobalClusterResult) {
    CreateGlobalClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalClusterResult || (CreateGlobalClusterResult = {}));
var GlobalClusterAlreadyExistsFault = (function (_super) {
    __extends(GlobalClusterAlreadyExistsFault, _super);
    function GlobalClusterAlreadyExistsFault(opts) {
        var _this = _super.call(this, __assign({ name: "GlobalClusterAlreadyExistsFault", $fault: "client" }, opts)) || this;
        _this.name = "GlobalClusterAlreadyExistsFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, GlobalClusterAlreadyExistsFault.prototype);
        return _this;
    }
    return GlobalClusterAlreadyExistsFault;
}(__BaseException));
export { GlobalClusterAlreadyExistsFault };
var GlobalClusterQuotaExceededFault = (function (_super) {
    __extends(GlobalClusterQuotaExceededFault, _super);
    function GlobalClusterQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "GlobalClusterQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "GlobalClusterQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, GlobalClusterQuotaExceededFault.prototype);
        return _this;
    }
    return GlobalClusterQuotaExceededFault;
}(__BaseException));
export { GlobalClusterQuotaExceededFault };
export var CreateOptionGroupMessage;
(function (CreateOptionGroupMessage) {
    CreateOptionGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateOptionGroupMessage || (CreateOptionGroupMessage = {}));
export var CreateOptionGroupResult;
(function (CreateOptionGroupResult) {
    CreateOptionGroupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateOptionGroupResult || (CreateOptionGroupResult = {}));
export var DeleteCustomAvailabilityZoneMessage;
(function (DeleteCustomAvailabilityZoneMessage) {
    DeleteCustomAvailabilityZoneMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCustomAvailabilityZoneMessage || (DeleteCustomAvailabilityZoneMessage = {}));
export var DeleteCustomAvailabilityZoneResult;
(function (DeleteCustomAvailabilityZoneResult) {
    DeleteCustomAvailabilityZoneResult.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.CustomAvailabilityZone && {
        CustomAvailabilityZone: CustomAvailabilityZone.filterSensitiveLog(obj.CustomAvailabilityZone),
    }))); };
})(DeleteCustomAvailabilityZoneResult || (DeleteCustomAvailabilityZoneResult = {}));
var CustomDBEngineVersionNotFoundFault = (function (_super) {
    __extends(CustomDBEngineVersionNotFoundFault, _super);
    function CustomDBEngineVersionNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "CustomDBEngineVersionNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "CustomDBEngineVersionNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomDBEngineVersionNotFoundFault.prototype);
        return _this;
    }
    return CustomDBEngineVersionNotFoundFault;
}(__BaseException));
export { CustomDBEngineVersionNotFoundFault };
export var DeleteCustomDBEngineVersionMessage;
(function (DeleteCustomDBEngineVersionMessage) {
    DeleteCustomDBEngineVersionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCustomDBEngineVersionMessage || (DeleteCustomDBEngineVersionMessage = {}));
var InvalidCustomDBEngineVersionStateFault = (function (_super) {
    __extends(InvalidCustomDBEngineVersionStateFault, _super);
    function InvalidCustomDBEngineVersionStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidCustomDBEngineVersionStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidCustomDBEngineVersionStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidCustomDBEngineVersionStateFault.prototype);
        return _this;
    }
    return InvalidCustomDBEngineVersionStateFault;
}(__BaseException));
export { InvalidCustomDBEngineVersionStateFault };
export var DeleteDBClusterMessage;
(function (DeleteDBClusterMessage) {
    DeleteDBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterMessage || (DeleteDBClusterMessage = {}));
export var DeleteDBClusterResult;
(function (DeleteDBClusterResult) {
    DeleteDBClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterResult || (DeleteDBClusterResult = {}));
var DBClusterEndpointNotFoundFault = (function (_super) {
    __extends(DBClusterEndpointNotFoundFault, _super);
    function DBClusterEndpointNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterEndpointNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterEndpointNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterEndpointNotFoundFault.prototype);
        return _this;
    }
    return DBClusterEndpointNotFoundFault;
}(__BaseException));
export { DBClusterEndpointNotFoundFault };
export var DeleteDBClusterEndpointMessage;
(function (DeleteDBClusterEndpointMessage) {
    DeleteDBClusterEndpointMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterEndpointMessage || (DeleteDBClusterEndpointMessage = {}));
var InvalidDBClusterEndpointStateFault = (function (_super) {
    __extends(InvalidDBClusterEndpointStateFault, _super);
    function InvalidDBClusterEndpointStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBClusterEndpointStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBClusterEndpointStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBClusterEndpointStateFault.prototype);
        return _this;
    }
    return InvalidDBClusterEndpointStateFault;
}(__BaseException));
export { InvalidDBClusterEndpointStateFault };
export var DeleteDBClusterParameterGroupMessage;
(function (DeleteDBClusterParameterGroupMessage) {
    DeleteDBClusterParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterParameterGroupMessage || (DeleteDBClusterParameterGroupMessage = {}));
var InvalidDBParameterGroupStateFault = (function (_super) {
    __extends(InvalidDBParameterGroupStateFault, _super);
    function InvalidDBParameterGroupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBParameterGroupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBParameterGroupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBParameterGroupStateFault.prototype);
        return _this;
    }
    return InvalidDBParameterGroupStateFault;
}(__BaseException));
export { InvalidDBParameterGroupStateFault };
export var DeleteDBClusterSnapshotMessage;
(function (DeleteDBClusterSnapshotMessage) {
    DeleteDBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterSnapshotMessage || (DeleteDBClusterSnapshotMessage = {}));
export var DeleteDBClusterSnapshotResult;
(function (DeleteDBClusterSnapshotResult) {
    DeleteDBClusterSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBClusterSnapshotResult || (DeleteDBClusterSnapshotResult = {}));
var DBInstanceAutomatedBackupQuotaExceededFault = (function (_super) {
    __extends(DBInstanceAutomatedBackupQuotaExceededFault, _super);
    function DBInstanceAutomatedBackupQuotaExceededFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceAutomatedBackupQuotaExceededFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceAutomatedBackupQuotaExceededFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceAutomatedBackupQuotaExceededFault.prototype);
        return _this;
    }
    return DBInstanceAutomatedBackupQuotaExceededFault;
}(__BaseException));
export { DBInstanceAutomatedBackupQuotaExceededFault };
export var DeleteDBInstanceMessage;
(function (DeleteDBInstanceMessage) {
    DeleteDBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceMessage || (DeleteDBInstanceMessage = {}));
export var DeleteDBInstanceResult;
(function (DeleteDBInstanceResult) {
    DeleteDBInstanceResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceResult || (DeleteDBInstanceResult = {}));
var DBInstanceAutomatedBackupNotFoundFault = (function (_super) {
    __extends(DBInstanceAutomatedBackupNotFoundFault, _super);
    function DBInstanceAutomatedBackupNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBInstanceAutomatedBackupNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBInstanceAutomatedBackupNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBInstanceAutomatedBackupNotFoundFault.prototype);
        return _this;
    }
    return DBInstanceAutomatedBackupNotFoundFault;
}(__BaseException));
export { DBInstanceAutomatedBackupNotFoundFault };
export var DeleteDBInstanceAutomatedBackupMessage;
(function (DeleteDBInstanceAutomatedBackupMessage) {
    DeleteDBInstanceAutomatedBackupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceAutomatedBackupMessage || (DeleteDBInstanceAutomatedBackupMessage = {}));
export var RestoreWindow;
(function (RestoreWindow) {
    RestoreWindow.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreWindow || (RestoreWindow = {}));
export var DBInstanceAutomatedBackup;
(function (DBInstanceAutomatedBackup) {
    DBInstanceAutomatedBackup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackup || (DBInstanceAutomatedBackup = {}));
export var DeleteDBInstanceAutomatedBackupResult;
(function (DeleteDBInstanceAutomatedBackupResult) {
    DeleteDBInstanceAutomatedBackupResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBInstanceAutomatedBackupResult || (DeleteDBInstanceAutomatedBackupResult = {}));
var InvalidDBInstanceAutomatedBackupStateFault = (function (_super) {
    __extends(InvalidDBInstanceAutomatedBackupStateFault, _super);
    function InvalidDBInstanceAutomatedBackupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBInstanceAutomatedBackupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBInstanceAutomatedBackupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBInstanceAutomatedBackupStateFault.prototype);
        return _this;
    }
    return InvalidDBInstanceAutomatedBackupStateFault;
}(__BaseException));
export { InvalidDBInstanceAutomatedBackupStateFault };
export var DeleteDBParameterGroupMessage;
(function (DeleteDBParameterGroupMessage) {
    DeleteDBParameterGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBParameterGroupMessage || (DeleteDBParameterGroupMessage = {}));
export var DeleteDBProxyRequest;
(function (DeleteDBProxyRequest) {
    DeleteDBProxyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyRequest || (DeleteDBProxyRequest = {}));
export var DeleteDBProxyResponse;
(function (DeleteDBProxyResponse) {
    DeleteDBProxyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyResponse || (DeleteDBProxyResponse = {}));
var DBProxyEndpointNotFoundFault = (function (_super) {
    __extends(DBProxyEndpointNotFoundFault, _super);
    function DBProxyEndpointNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyEndpointNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyEndpointNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyEndpointNotFoundFault.prototype);
        return _this;
    }
    return DBProxyEndpointNotFoundFault;
}(__BaseException));
export { DBProxyEndpointNotFoundFault };
export var DeleteDBProxyEndpointRequest;
(function (DeleteDBProxyEndpointRequest) {
    DeleteDBProxyEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyEndpointRequest || (DeleteDBProxyEndpointRequest = {}));
export var DeleteDBProxyEndpointResponse;
(function (DeleteDBProxyEndpointResponse) {
    DeleteDBProxyEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBProxyEndpointResponse || (DeleteDBProxyEndpointResponse = {}));
var InvalidDBProxyEndpointStateFault = (function (_super) {
    __extends(InvalidDBProxyEndpointStateFault, _super);
    function InvalidDBProxyEndpointStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBProxyEndpointStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBProxyEndpointStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBProxyEndpointStateFault.prototype);
        return _this;
    }
    return InvalidDBProxyEndpointStateFault;
}(__BaseException));
export { InvalidDBProxyEndpointStateFault };
export var DeleteDBSecurityGroupMessage;
(function (DeleteDBSecurityGroupMessage) {
    DeleteDBSecurityGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSecurityGroupMessage || (DeleteDBSecurityGroupMessage = {}));
export var DeleteDBSnapshotMessage;
(function (DeleteDBSnapshotMessage) {
    DeleteDBSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSnapshotMessage || (DeleteDBSnapshotMessage = {}));
export var DeleteDBSnapshotResult;
(function (DeleteDBSnapshotResult) {
    DeleteDBSnapshotResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSnapshotResult || (DeleteDBSnapshotResult = {}));
export var DeleteDBSubnetGroupMessage;
(function (DeleteDBSubnetGroupMessage) {
    DeleteDBSubnetGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDBSubnetGroupMessage || (DeleteDBSubnetGroupMessage = {}));
var InvalidDBSubnetStateFault = (function (_super) {
    __extends(InvalidDBSubnetStateFault, _super);
    function InvalidDBSubnetStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidDBSubnetStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidDBSubnetStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidDBSubnetStateFault.prototype);
        return _this;
    }
    return InvalidDBSubnetStateFault;
}(__BaseException));
export { InvalidDBSubnetStateFault };
export var DeleteEventSubscriptionMessage;
(function (DeleteEventSubscriptionMessage) {
    DeleteEventSubscriptionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEventSubscriptionMessage || (DeleteEventSubscriptionMessage = {}));
export var DeleteEventSubscriptionResult;
(function (DeleteEventSubscriptionResult) {
    DeleteEventSubscriptionResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEventSubscriptionResult || (DeleteEventSubscriptionResult = {}));
var InvalidEventSubscriptionStateFault = (function (_super) {
    __extends(InvalidEventSubscriptionStateFault, _super);
    function InvalidEventSubscriptionStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidEventSubscriptionStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidEventSubscriptionStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidEventSubscriptionStateFault.prototype);
        return _this;
    }
    return InvalidEventSubscriptionStateFault;
}(__BaseException));
export { InvalidEventSubscriptionStateFault };
export var DeleteGlobalClusterMessage;
(function (DeleteGlobalClusterMessage) {
    DeleteGlobalClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGlobalClusterMessage || (DeleteGlobalClusterMessage = {}));
export var DeleteGlobalClusterResult;
(function (DeleteGlobalClusterResult) {
    DeleteGlobalClusterResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGlobalClusterResult || (DeleteGlobalClusterResult = {}));
export var DeleteInstallationMediaMessage;
(function (DeleteInstallationMediaMessage) {
    DeleteInstallationMediaMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteInstallationMediaMessage || (DeleteInstallationMediaMessage = {}));
export var InstallationMediaFailureCause;
(function (InstallationMediaFailureCause) {
    InstallationMediaFailureCause.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstallationMediaFailureCause || (InstallationMediaFailureCause = {}));
export var InstallationMedia;
(function (InstallationMedia) {
    InstallationMedia.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstallationMedia || (InstallationMedia = {}));
var InstallationMediaNotFoundFault = (function (_super) {
    __extends(InstallationMediaNotFoundFault, _super);
    function InstallationMediaNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "InstallationMediaNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "InstallationMediaNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InstallationMediaNotFoundFault.prototype);
        return _this;
    }
    return InstallationMediaNotFoundFault;
}(__BaseException));
export { InstallationMediaNotFoundFault };
export var DeleteOptionGroupMessage;
(function (DeleteOptionGroupMessage) {
    DeleteOptionGroupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteOptionGroupMessage || (DeleteOptionGroupMessage = {}));
var InvalidOptionGroupStateFault = (function (_super) {
    __extends(InvalidOptionGroupStateFault, _super);
    function InvalidOptionGroupStateFault(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidOptionGroupStateFault", $fault: "client" }, opts)) || this;
        _this.name = "InvalidOptionGroupStateFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidOptionGroupStateFault.prototype);
        return _this;
    }
    return InvalidOptionGroupStateFault;
}(__BaseException));
export { InvalidOptionGroupStateFault };
var DBProxyTargetNotFoundFault = (function (_super) {
    __extends(DBProxyTargetNotFoundFault, _super);
    function DBProxyTargetNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBProxyTargetNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBProxyTargetNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBProxyTargetNotFoundFault.prototype);
        return _this;
    }
    return DBProxyTargetNotFoundFault;
}(__BaseException));
export { DBProxyTargetNotFoundFault };
export var DeregisterDBProxyTargetsRequest;
(function (DeregisterDBProxyTargetsRequest) {
    DeregisterDBProxyTargetsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeregisterDBProxyTargetsRequest || (DeregisterDBProxyTargetsRequest = {}));
export var DeregisterDBProxyTargetsResponse;
(function (DeregisterDBProxyTargetsResponse) {
    DeregisterDBProxyTargetsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeregisterDBProxyTargetsResponse || (DeregisterDBProxyTargetsResponse = {}));
export var DescribeAccountAttributesMessage;
(function (DescribeAccountAttributesMessage) {
    DescribeAccountAttributesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccountAttributesMessage || (DescribeAccountAttributesMessage = {}));
export var Certificate;
(function (Certificate) {
    Certificate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Certificate || (Certificate = {}));
export var CertificateMessage;
(function (CertificateMessage) {
    CertificateMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CertificateMessage || (CertificateMessage = {}));
var CertificateNotFoundFault = (function (_super) {
    __extends(CertificateNotFoundFault, _super);
    function CertificateNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "CertificateNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "CertificateNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CertificateNotFoundFault.prototype);
        return _this;
    }
    return CertificateNotFoundFault;
}(__BaseException));
export { CertificateNotFoundFault };
export var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filter || (Filter = {}));
export var DescribeCertificatesMessage;
(function (DescribeCertificatesMessage) {
    DescribeCertificatesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCertificatesMessage || (DescribeCertificatesMessage = {}));
export var CustomAvailabilityZoneMessage;
(function (CustomAvailabilityZoneMessage) {
    CustomAvailabilityZoneMessage.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.CustomAvailabilityZones && {
        CustomAvailabilityZones: obj.CustomAvailabilityZones.map(function (item) {
            return CustomAvailabilityZone.filterSensitiveLog(item);
        }),
    }))); };
})(CustomAvailabilityZoneMessage || (CustomAvailabilityZoneMessage = {}));
export var DescribeCustomAvailabilityZonesMessage;
(function (DescribeCustomAvailabilityZonesMessage) {
    DescribeCustomAvailabilityZonesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCustomAvailabilityZonesMessage || (DescribeCustomAvailabilityZonesMessage = {}));
export var DBClusterBacktrackMessage;
(function (DBClusterBacktrackMessage) {
    DBClusterBacktrackMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterBacktrackMessage || (DBClusterBacktrackMessage = {}));
var DBClusterBacktrackNotFoundFault = (function (_super) {
    __extends(DBClusterBacktrackNotFoundFault, _super);
    function DBClusterBacktrackNotFoundFault(opts) {
        var _this = _super.call(this, __assign({ name: "DBClusterBacktrackNotFoundFault", $fault: "client" }, opts)) || this;
        _this.name = "DBClusterBacktrackNotFoundFault";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DBClusterBacktrackNotFoundFault.prototype);
        return _this;
    }
    return DBClusterBacktrackNotFoundFault;
}(__BaseException));
export { DBClusterBacktrackNotFoundFault };
export var DescribeDBClusterBacktracksMessage;
(function (DescribeDBClusterBacktracksMessage) {
    DescribeDBClusterBacktracksMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterBacktracksMessage || (DescribeDBClusterBacktracksMessage = {}));
export var DBClusterEndpointMessage;
(function (DBClusterEndpointMessage) {
    DBClusterEndpointMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterEndpointMessage || (DBClusterEndpointMessage = {}));
export var DescribeDBClusterEndpointsMessage;
(function (DescribeDBClusterEndpointsMessage) {
    DescribeDBClusterEndpointsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterEndpointsMessage || (DescribeDBClusterEndpointsMessage = {}));
export var DBClusterParameterGroupsMessage;
(function (DBClusterParameterGroupsMessage) {
    DBClusterParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterParameterGroupsMessage || (DBClusterParameterGroupsMessage = {}));
export var DescribeDBClusterParameterGroupsMessage;
(function (DescribeDBClusterParameterGroupsMessage) {
    DescribeDBClusterParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterParameterGroupsMessage || (DescribeDBClusterParameterGroupsMessage = {}));
export var Parameter;
(function (Parameter) {
    Parameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Parameter || (Parameter = {}));
export var DBClusterParameterGroupDetails;
(function (DBClusterParameterGroupDetails) {
    DBClusterParameterGroupDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterParameterGroupDetails || (DBClusterParameterGroupDetails = {}));
export var DescribeDBClusterParametersMessage;
(function (DescribeDBClusterParametersMessage) {
    DescribeDBClusterParametersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterParametersMessage || (DescribeDBClusterParametersMessage = {}));
export var DBClusterMessage;
(function (DBClusterMessage) {
    DBClusterMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterMessage || (DBClusterMessage = {}));
export var DescribeDBClustersMessage;
(function (DescribeDBClustersMessage) {
    DescribeDBClustersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClustersMessage || (DescribeDBClustersMessage = {}));
export var DescribeDBClusterSnapshotAttributesMessage;
(function (DescribeDBClusterSnapshotAttributesMessage) {
    DescribeDBClusterSnapshotAttributesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterSnapshotAttributesMessage || (DescribeDBClusterSnapshotAttributesMessage = {}));
export var DBClusterSnapshotAttribute;
(function (DBClusterSnapshotAttribute) {
    DBClusterSnapshotAttribute.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotAttribute || (DBClusterSnapshotAttribute = {}));
export var DBClusterSnapshotAttributesResult;
(function (DBClusterSnapshotAttributesResult) {
    DBClusterSnapshotAttributesResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotAttributesResult || (DBClusterSnapshotAttributesResult = {}));
export var DescribeDBClusterSnapshotAttributesResult;
(function (DescribeDBClusterSnapshotAttributesResult) {
    DescribeDBClusterSnapshotAttributesResult.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterSnapshotAttributesResult || (DescribeDBClusterSnapshotAttributesResult = {}));
export var DBClusterSnapshotMessage;
(function (DBClusterSnapshotMessage) {
    DBClusterSnapshotMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBClusterSnapshotMessage || (DBClusterSnapshotMessage = {}));
export var DescribeDBClusterSnapshotsMessage;
(function (DescribeDBClusterSnapshotsMessage) {
    DescribeDBClusterSnapshotsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBClusterSnapshotsMessage || (DescribeDBClusterSnapshotsMessage = {}));
export var DBEngineVersionMessage;
(function (DBEngineVersionMessage) {
    DBEngineVersionMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBEngineVersionMessage || (DBEngineVersionMessage = {}));
export var DescribeDBEngineVersionsMessage;
(function (DescribeDBEngineVersionsMessage) {
    DescribeDBEngineVersionsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBEngineVersionsMessage || (DescribeDBEngineVersionsMessage = {}));
export var DBInstanceAutomatedBackupMessage;
(function (DBInstanceAutomatedBackupMessage) {
    DBInstanceAutomatedBackupMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceAutomatedBackupMessage || (DBInstanceAutomatedBackupMessage = {}));
export var DescribeDBInstanceAutomatedBackupsMessage;
(function (DescribeDBInstanceAutomatedBackupsMessage) {
    DescribeDBInstanceAutomatedBackupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBInstanceAutomatedBackupsMessage || (DescribeDBInstanceAutomatedBackupsMessage = {}));
export var DBInstanceMessage;
(function (DBInstanceMessage) {
    DBInstanceMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBInstanceMessage || (DBInstanceMessage = {}));
export var DescribeDBInstancesMessage;
(function (DescribeDBInstancesMessage) {
    DescribeDBInstancesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBInstancesMessage || (DescribeDBInstancesMessage = {}));
export var DescribeDBLogFilesMessage;
(function (DescribeDBLogFilesMessage) {
    DescribeDBLogFilesMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBLogFilesMessage || (DescribeDBLogFilesMessage = {}));
export var DescribeDBLogFilesDetails;
(function (DescribeDBLogFilesDetails) {
    DescribeDBLogFilesDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBLogFilesDetails || (DescribeDBLogFilesDetails = {}));
export var DescribeDBLogFilesResponse;
(function (DescribeDBLogFilesResponse) {
    DescribeDBLogFilesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBLogFilesResponse || (DescribeDBLogFilesResponse = {}));
export var DBParameterGroupsMessage;
(function (DBParameterGroupsMessage) {
    DBParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupsMessage || (DBParameterGroupsMessage = {}));
export var DescribeDBParameterGroupsMessage;
(function (DescribeDBParameterGroupsMessage) {
    DescribeDBParameterGroupsMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBParameterGroupsMessage || (DescribeDBParameterGroupsMessage = {}));
export var DBParameterGroupDetails;
(function (DBParameterGroupDetails) {
    DBParameterGroupDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBParameterGroupDetails || (DBParameterGroupDetails = {}));
export var DescribeDBParametersMessage;
(function (DescribeDBParametersMessage) {
    DescribeDBParametersMessage.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBParametersMessage || (DescribeDBParametersMessage = {}));
export var DescribeDBProxiesRequest;
(function (DescribeDBProxiesRequest) {
    DescribeDBProxiesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxiesRequest || (DescribeDBProxiesRequest = {}));
export var DescribeDBProxiesResponse;
(function (DescribeDBProxiesResponse) {
    DescribeDBProxiesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxiesResponse || (DescribeDBProxiesResponse = {}));
export var DescribeDBProxyEndpointsRequest;
(function (DescribeDBProxyEndpointsRequest) {
    DescribeDBProxyEndpointsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyEndpointsRequest || (DescribeDBProxyEndpointsRequest = {}));
export var DescribeDBProxyEndpointsResponse;
(function (DescribeDBProxyEndpointsResponse) {
    DescribeDBProxyEndpointsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyEndpointsResponse || (DescribeDBProxyEndpointsResponse = {}));
export var DescribeDBProxyTargetGroupsRequest;
(function (DescribeDBProxyTargetGroupsRequest) {
    DescribeDBProxyTargetGroupsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyTargetGroupsRequest || (DescribeDBProxyTargetGroupsRequest = {}));
export var ConnectionPoolConfigurationInfo;
(function (ConnectionPoolConfigurationInfo) {
    ConnectionPoolConfigurationInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionPoolConfigurationInfo || (ConnectionPoolConfigurationInfo = {}));
export var DBProxyTargetGroup;
(function (DBProxyTargetGroup) {
    DBProxyTargetGroup.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DBProxyTargetGroup || (DBProxyTargetGroup = {}));
export var DescribeDBProxyTargetGroupsResponse;
(function (DescribeDBProxyTargetGroupsResponse) {
    DescribeDBProxyTargetGroupsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyTargetGroupsResponse || (DescribeDBProxyTargetGroupsResponse = {}));
export var DescribeDBProxyTargetsRequest;
(function (DescribeDBProxyTargetsRequest) {
    DescribeDBProxyTargetsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDBProxyTargetsRequest || (DescribeDBProxyTargetsRequest = {}));
